File: app/src/main/java/com/smile/watchmovie/adapter/EpisodeAdapter.java
package com.smile.watchmovie.adapter;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.smile.watchmovie.R;
import com.smile.watchmovie.activity.WatchFilmActivity;
import com.smile.watchmovie.databinding.ItemEpisodeBinding;
import com.smile.watchmovie.fragment.IntroduceFilmFragment;
import com.smile.watchmovie.model.SubFilm;

import java.util.List;

public class EpisodeAdapter extends RecyclerView.Adapter<EpisodeAdapter.EpisodeViewHolder> {

    public List<SubFilm> subFilmList;
    public final WatchFilmActivity mWatchFilmActivity;
    public final IntroduceFilmFragment mIntroduceFilmFragment;

    public EpisodeAdapter(WatchFilmActivity watchFilmActivity, IntroduceFilmFragment mIntroduceFilmFragment) {
        this.mWatchFilmActivity = watchFilmActivity;
        this.mIntroduceFilmFragment = mIntroduceFilmFragment;
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setData(List<SubFilm> subVideoList) {
        this.subFilmList = subVideoList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public EpisodeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        return new EpisodeViewHolder(ItemEpisodeBinding.inflate(inflater, parent, false));
    }

    @SuppressLint({"NotifyDataSetChanged", "SetTextI18n"})
    @Override
    public void onBindViewHolder(@NonNull EpisodeViewHolder holder, @SuppressLint("RecyclerView") int position) {
        SubFilm subFilm = subFilmList.get(position);
        if (subFilm == null) {
            return;
        }
        holder.itemEpisodeBinding.tvEpisode.setText(mWatchFilmActivity.getString(R.string.episode_film, subFilm.getEpisode()));
        holder.itemEpisodeBinding.layoutEpisode.setOnClickListener(v ->{
            mWatchFilmActivity.playVideo(subFilm);
            mIntroduceFilmFragment.episodeFilmPlaying(subFilm);
        });
        if(subFilm.getIsWatching()){
            holder.itemEpisodeBinding.tvEpisode.setBackgroundResource(R.drawable.botron_background_click);
            holder.itemEpisodeBinding.tvEpisode.setTextColor(Color.parseColor("#1877F2"));
        }
        else{
            holder.itemEpisodeBinding.tvEpisode.setBackgroundResource(R.drawable.botron_background);
            holder.itemEpisodeBinding.tvEpisode.setTextColor(Color.parseColor("#000000"));
        }
    }

    @Override
    public int getItemCount() {
        if (subFilmList != null)
            return subFilmList.size();
        return 0;
    }

    public static class EpisodeViewHolder extends RecyclerView.ViewHolder {
        private final ItemEpisodeBinding itemEpisodeBinding;
        public EpisodeViewHolder(ItemEpisodeBinding itemEpisodeBinding) {
            super(itemEpisodeBinding.getRoot());
            this.itemEpisodeBinding = itemEpisodeBinding;
        }
    }
}

File: app/src/main/java/com/smile/watchmovie/adapter/EpisodeAdapter.kt
package com.smile.watchmovie.adapter

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.smile.watchmovie.R
import com.smile.watchmovie.databinding.ItemEpisodeBinding
import com.smile.watchmovie.model.SubFilm

class EpisodeAdapter(private val listener: OnListener) :
    RecyclerView.Adapter<EpisodeAdapter.ViewHolder>() {
    private var subFilmList = ArrayList<SubFilm>()

    @SuppressLint("NotifyDataSetChanged")
    fun setData(newSubVideoList: List<SubFilm>) {
        subFilmList = newSubVideoList as ArrayList<SubFilm>
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemEpisodeBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        with(holder) {
            with(subFilmList[position]) {
                binding.apply {
                    tvEpisode.apply {
                        text = context.getString(R.string.episode_film, this@with.episode)
                        setBackgroundResource(
                            if (this@with.isWatching) {
                                R.drawable.bg_sub_film_clicked
                            } else {
                                R.drawable.bg_corner_dialog
                            }
                        )
                    }

                    root.setOnClickListener {
                        listener.onClickSubFilm(this@with)
                    }
                }
            }
        }
    }

    override fun getItemCount() = subFilmList.size

    inner class ViewHolder(val binding: ItemEpisodeBinding) : RecyclerView.ViewHolder(binding.root)

    interface OnListener {
        fun onClickSubFilm(subFilm: SubFilm)
    }
}